lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -31,12 +31,17 @@
         required: true
     source:
         description:
-            - OS disk source from the same region, including a virtual machine id or name,
-              OS disk blob uri, managed OS disk id or name, or OS snapshot id or name.
+            - OS disk source from the same region.
+            - It can be a virtual machine, OS disk blob uri, managed OS disk, or OS snapshot.
+            - Each type of source expect for blob uri can be given as resource id, name or a dict contains C(resource_group), C(name) and C(types).
```

Comment:
it's not so straightforward, maybe some samples?

Reply:
add an example